src/core/solvers/ddp.cpp

```diff
@@ -335,6 +285,68 @@ void SolverDDP::forwardPass(const double steplength) {
   STOP_PROFILER("SolverDDP::forwardPass");
 }
 
+void SolverDDP::computeHamiltonianFunction(const std::size_t t, const boost::shared_ptr<ActionModelAbstract>& model,
+                                           const boost::shared_ptr<ActionDataAbstract>& data) {
+  const std::size_t nu = model->get_nu();
+  const Eigen::MatrixXd& Vxx_p = Vxx_[t + 1];
```

Comment:
I understand this request, however, I am not sure if we should do it. This will add a lot of checks that are not needed when the user runs the solver normally. We could definetely add an assert but this won't be useful for Python developers.

I am not sure what to do here. What is the opinion of others?